rust/lance/src/dataset.rs

```diff
@@ -1530,6 +1528,32 @@ impl Dataset {
         Ok(())
     }
 
+    /// Fragment-level processing interface for distributed execution
+    /// Enables external frameworks like Ray to parallelize fragment processing
+    async fn process_fragments(
+        &self,
+        fragment_ids: Vec<usize>,
+        left_on: &str,
+        joiner: Arc<HashJoiner>,
+    ) -> Result<Vec<Fragment>> {
+        let fragments = stream::iter(fragment_ids)
+            .map(|fragment_id| async move {
+                self.get_fragment(fragment_id).ok_or_else(|| {
+                    Error::invalid_input(format!("Fragment {} not found", fragment_id), location!())
+                })
+            })
+            .buffered(2) // 缓冲最多2个并发请求
```

Comment:
Why 2?  I think a higher number might be better?  Seems like the tradeoff is...

* Higher number is more RAM but faster?